Source/Tests/HarnessHelpers.h

```diff
@@ -390,8 +390,8 @@ namespace FEX::HarnessHelper {
 
     bool MapMemory(const MapperFn& Mapper, const UnmapperFn& Unmapper) override {
       bool LimitedSize = true;
-      auto DoMMap = [](uint64_t Address, size_t Size) -> void* {
-        void *Result = FEXCore::Allocator::mmap(reinterpret_cast<void*>(Address), Size, PROT_READ | PROT_WRITE | PROT_EXEC, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+      auto DoMMap = [Mapper](uint64_t Address, size_t Size) -> void* {
```

Comment:
Interesting, I thought std::function always does heap allocations when capturing.

Though, I'd rather keep it as is, for consistency with the ElfCodeLoader. We can change both of them in a follow up.

Reply:
> Interesting, I thought std::function always does heap allocations when capturing.

The original `std::function` will be heap-allocated, but taking references avoids the *additional* heap-allocation that would be incurred by copying it.

Another option is to *move* the function into the captures:
```suggestion
      auto DoMMap = [Mapper=std::move(Mapper)](uint64_t Address, size_t Size) -> void* {
```

> for consistency with the ElfCodeLoader

I don't think consistent use of pessimized code is a desirable quality. Also, this seems to be a weird module to be consistent with?

Reply:
> The original std::function will be heap-allocated, but taking references avoids the additional heap-allocation that would be incurred by copying it.

Oh right, it'll lead to a copy of the Mapper.

> I don't think consistent use of pessimized code is a desirable quality. Also, this seems to be a weird module to be consistent with?

This part of the code (and other large paths) are copy pasted between the two loaders. This bug seems to have happened because we failed to propagate changes from one to the other, actually.

Reply:
(Switched both codepaths to use refs)